src/lib.rs

```diff
@@ -541,10 +533,11 @@ pub fn session_channel<P: HasDual>() -> (Chan<(), P>, Chan<(), P::Dual>) {
 
 /// Connect two functions using a session typed channel.
 pub fn connect<F1, F2, P>(srv: F1, cli: F2)
-    where F1: Fn(Chan<(), P>) + marker::Send + 'static,
-          F2: Fn(Chan<(), P::Dual>) + marker::Send,
-          P: HasDual + marker::Send + 'static,
-          <P as HasDual>::Dual: HasDual + marker::Send + 'static
+where
+    F1: Fn(Chan<(), P>) + marker::Send + 'static,
+    F2: Fn(Chan<(), P::Dual>) + marker::Send,
+    P: HasDual + marker::Send + 'static,
+    <P as HasDual>::Dual: HasDual + marker::Send + 'static,
```

Comment:
This line can now be be `P::Dual: HasDual + marker::Send + 'static` (no need for the cast anymore)